### ISSUE TITLE
chronos: find a cleaner way to support traditional statistical style forecast model

### ISSUE THREAD

**Issue by shane-huang** [intel-analytics/analytics-zoo#4126]
There're two style of forecasting model, we need a cleaner way to support traiditional statistical style models, as they don't require X input in predict and evaluate. 

Regular Regression style (e.g. tcn, seq2seq, lstm, xgboost)
   - Train: [X,y] pairs rolled from consecutive time series
   - Predict: Last X steps -> y future steps
   - Lookback is a tunable hyper parameter
   - regular regression models can be used this way
   - Needs to keep a user side buffer to use on streaming input
Traditional Statistical style (tcmf, arima, prophet)
   - Predict: None -> y future steps immediately after the last X in train
   - Lookback may not be tunable 
   - if model is fit on train set only, user will need an extra step to fit on train+validation set before use it on test. 
   - Hard to handle non-consecutive inputs, needs to keep track of the last timestamp.
